src/app/header/header-profile/facebook-button/facebook-button.component.ts

```diff
@@ -0,0 +1,80 @@
+import { Component, OnInit } from '@angular/core';
+import { AuthService } from '../../../services/auth.service';
+import { User } from '../../../models/user.model';
+declare var FB: any;
+
+@Component({
+  selector: 'app-facebook-button',
+  templateUrl: './facebook-button.component.html',
+  styleUrls: ['./facebook-button.component.scss']
+})
+export class FacebookButtonComponent implements OnInit {
+
+  constructor(private authService: AuthService) { }
+
+  ngOnInit(): void {
+    this.initFacebookLogin();
+  }
+
+  private initFacebookLogin(): void {
+    (window as any).fbAsyncInit = () => {
+      FB.init({
+        appId      : '408423610210198',
+        cookie     : true,
+        xfbml      : true,
+        version    : 'v3.1'
+      });
+      FB.AppEvents.logPageView();
+    };
+
+    (((d, s, id): void => {
```

Comment:
What is it ?
`(((d, s, id): void => {`

Reply:
it is an origin code from quick start tutorial by facebook

Reply:
![image](https://user-images.githubusercontent.com/64072558/102875466-7ff68e00-444c-11eb-8562-90c0ba1f26dc.png)